internal/statesync/dispatcher.go

```diff
@@ -193,47 +106,100 @@ func (d *dispatcher) dispatch(peer types.NodeID, height int64) (chan *types.Ligh
 			Height: uint64(height),
 		},
 	}
+
 	return ch, nil
 }
 
-// release appends the peer back to the list and deletes the allocated call so
-// that a new call can be made to that peer
-func (d *dispatcher) release(peer types.NodeID) {
+// respond allows the underlying process which receives requests on the
+// requestCh to respond with the respective light block
+func (d *Dispatcher) Respond(lb *proto.LightBlock, peer types.NodeID) error {
 	d.mtx.Lock()
 	defer d.mtx.Unlock()
-	if call, ok := d.calls[peer]; ok {
-		close(call)
+
+	// check that the response came from a request
+	answerCh, ok := d.calls[peer]
+	if !ok {
+		// this can also happen if the response came in after the timeout
+		return errUnsolicitedResponse
+	}
+
+	if lb == nil {
```

Comment:
So sending nil light blocks is a pattern that I completely made up to signal to the receiver that the sender doesn't have the requested resource. I was thinking (and becoming more convinced) that it may be more appropriate to have an accompanying enum field that does this.